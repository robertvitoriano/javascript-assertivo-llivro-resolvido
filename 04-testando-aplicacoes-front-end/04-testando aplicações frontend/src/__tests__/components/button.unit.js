import {screen} from "@testing-library/react"
import userEvent from '@testing-library/user-event'
import Button from '../../components/button'
import { renderWithTheme } from "../../testUtils"
describe("<Button/>",()=>{
  it("Renderiza um botão corretamente", ()=>{
    renderWithTheme(<Button >something</Button>)
    const button =  screen.getByRole("button")
    expect(button).toBeInTheDocument()
    expect(button).toMatchSnapshot()
  })
  it("fires some click actions once it receives a function as prop",()=>{
    const testFunction = jest.fn()
    renderWithTheme(<Button onClick={testFunction}>Some text</Button>)
    const button = screen.getByRole("button")
    userEvent.click(button)
    expect(testFunction).toBeCalledTimes(1)    
  })
  
  it("Pode renderizar um ícone",()=>{
    const icon = <span>Some Icon</span>
    renderWithTheme(<Button icon={icon}>Some text</Button>)
    const iconFound = screen.getByText('Some Icon')
    expect(iconFound).toBeInTheDocument()
  })
})

//página 250
