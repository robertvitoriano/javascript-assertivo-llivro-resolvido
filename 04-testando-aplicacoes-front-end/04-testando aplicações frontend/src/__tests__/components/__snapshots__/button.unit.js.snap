// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Renderiza um bot√£o corretamente 1`] = `
<button
  class="sc-bdfBwQ iOlsKH"
  type=""
>
  <span>
    something
  </span>
</button>
`;
